src/Features/Core/Portable/Completion/Providers/AbstractSymbolCompletionProvider.cs

```diff
@@ -86,14 +92,23 @@ private bool ShouldIncludeInTargetTypedCompletionList(ISymbol symbol, ImmutableA
                 return false;
             }
 
+            type = type.WithoutNullability();
+
+            if (typeConvertibilityCache.TryGetValue(type, out var isConvertible))
+            {
+                return isConvertible;
+            }
+
             foreach (var inferredType in inferredTypes)
             {
-                if (semanticModel.Compilation.ClassifyCommonConversion(type.WithoutNullability(), inferredType.WithoutNullability()).IsImplicit)
+                if (semanticModel.Compilation.ClassifyCommonConversion(type, inferredType).IsImplicit)
                 {
+                    typeConvertibilityCache[type] = true;
                     return true;
                 }
             }
 
+            typeConvertibilityCache[type] = false;
             return false;
```

Comment:
i personally don't like the double assignment, and redundancy here.  Can we extract out the foreach loop into a helper function that just gives back the answer?  Then just assign the answer into the map and return it.

Reply:
Agreed. I've fixed it up, please let me know if it matches your expectations.